packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -137,6 +137,7 @@ export class TestObjectProvider {
     private _urlResolver: IUrlResolver | undefined;
     private _logger: ITelemetryBaseLogger | undefined;
     private readonly _documentIdStrategy: IDocumentIdStrategy;
+    private readonly _usedDocumentIds: Set<string>;
```

Comment:
A set is not needed as the test provider only supports one documentId at a time. A flag should be enough for this which is set when a container is created.
Maybe it can be built into documentIdStrategy but I don't see a simple way to do this.

Reply:
Also, this state has to be reset when `reset` is called on the test provider.